### ISSUE TITLE
Flux sometimes fails with timeout even though successfully deployed

### ISSUE THREAD

**Issue by gnobre** [weaveworks/flux#1162]
After running `fluxtctl release`, sometimes the release "fails" with a timeout even though it was actually successfully deployed

`$ fluxctl release --namespace=$NAMESPACE --controller=deployment/$SERVICE_NAME --update-image=$CONTAINER_VERSIONED_IMAGE --url=$FLUX_URL`
```
Submitting release ...
CONTROLLER                           STATUS   UPDATES
testing:deployment/proejct         success  registry.gitlab
Commit pushed:	d36a684
Error: timeout
Run 'fluxctl release --help' for usage.
ERROR: Job failed: exit status 1
```

This happens from time to time and the release itself is always successful, besides the job failed error. Retrying one or two times makes the error go away.

**Comment by squaremo** [comment]
Just to check, the last bit is GitLab reporting that its job failed, is that right?

The command `fluxctl release` goes through these steps:
 1. submit the release to fluxd and get a job ID back
 2. ask about the job, until it's complete (a commit is pushed by fluxd, and the sha1 is returned)
 3. ask about the commit until it's synced.

It's the last bit that is timing out, by the look of it; but really, the release was successful at the point the commit was pushed to the git repo. So, off the top of my head, we could print a warning to stderr and `exit(0)` if it times out waiting for the sync (you can now issue `fluxctl sync` to interactively trigger a sync). Then your gitlab job wouldn't look like it failed when it effectively succeeded. WDYT?

**Comment by gnobre** [comment]
Yes, this is running on a GitLab worker :+1: 

That would do it, the issue is really the job looking like it failed and we receiving a bunch of "failed pipeline" messages on slack and such when everything went just fine. Thank you!

**Comment by squaremo** [comment]
Closed in #1163.
